cluster/src/main/scala/org/apache/spark/executor/SnappyExecutor.scala

```diff
@@ -108,6 +108,24 @@ class SnappyExecutor(
       }
     }
   }
+
+  override def isStoreCloseException(t : Throwable) : Boolean = {
+    try {
+      Misc.checkIfCacheClosing(t)
+    } catch {
+      case ex: CacheClosedException => true
```

Comment:
This should catch the base "CancelException" and not the specific CacheClosedException.